structure_tests/utils.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"syscall"
+	"testing"
+)
+
+// given a list of environment variable key/value pairs, set these in the current environment.
+// also, keep track of the previous values of these vars to reset after test execution.
+func SetEnvVars(t *testing.T, vars []EnvVar) []EnvVar {
+	var originalVars []EnvVar
+	for _, env_var := range vars {
+		originalVars = append(originalVars, EnvVar{env_var.Key, os.Getenv(env_var.Key)})
+		if err := os.Setenv(env_var.Key, os.ExpandEnv(env_var.Value)); err != nil {
+			t.Fatalf("error setting env var: %s", err)
+		}
+	}
+	return originalVars
+}
+
+func ResetEnvVars(t *testing.T, vars []EnvVar) {
+	for _, env_var := range vars {
+		var err error
+		if env_var.Value == "" {
+			// if the previous value was empty string, the variable did not
+			// exist in the environment; unset it
+			err = os.Unsetenv(env_var.Key)
+		} else {
+			// otherwise, set it back to its previous value
+			err = os.Setenv(env_var.Key, env_var.Value)
+		}
+		if err != nil {
+			t.Fatalf("error resetting env var: %s", err)
+		}
+	}
+}
+
+func compileAndRunRegex(regex string, base string, t *testing.T, err string, shouldMatch bool) {
+	r, rErr := regexp.Compile(regex)
+	if rErr != nil {
+		t.Errorf("Error compiling regex %s : %s", regex, rErr.Error())
+		return
+	}
+	if shouldMatch != r.MatchString(base) {
+		t.Errorf(err)
+	}
+}
+
+// given an array of command parts, construct a full command and execute it against the
+// current environment. a list of environment variables can be passed to be set in the
+// environment before the command is executed. additionally, a boolean flag is passed
+// to specify whether or not we care about the output of the command.
+func ProcessCommand(t *testing.T, envVars []EnvVar, fullCommand []string,
```

Comment:
What's the different between this and processCommandv1?

Reply:
whoops, meant to get rid of the different versions of this method. fixed